SKAToolKit/SKATiledMap/SKATiledMap.m

```diff
@@ -772,20 +762,31 @@ - (void)cullAroundIndexX:(NSInteger)x
             endingY = 0;
         }
 
+        NSMutableArray *spritesToTrash = [[NSMutableArray alloc] init]; //array to hold sprites that have moved off-screen
+        [self.visibleArray
+            enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
```

Comment:
Even better, you could just check the 4 corner tiles, and if one of them moved off screen, remove all the tiles for that row, column or both, based on what direction the map was moving. Then you'd check in one spot. The logic will be more complicated, but would be way more efficient. Might have to restructure the `visibleSprites` to make it easy to know what the corner sprites are as well.